File: LeetCode/Unique Length-3 Palindromic Subsequences/main.kt
class Solution {
    fun countPalindromicSubsequence(s: String): Int {
        val index = Array(26) { arrayOf(-1, -1) }
        for (i in s.indices) {
            val curr = s[i] - 'a'
            if (index[curr][0] == -1)
                index[curr][0] = i
            else
                index[curr][1] = i
        }
        var ans = 0
        for (i in 0 until 26) {
            val curr = HashSet<Char>()
            for (j in index[i][0] + 1 until index[i][1]) {
                curr.add(s[j])
            }
            ans += curr.size
        }
        return ans
    }
}

File: LeetCode/Unique Length-3 Palindromic Subsequences/main.cpp
class Solution {
   public:
    int countPalindromicSubsequence(string s) {
        vector<pair<int, int>> index(26, {-1, -1});
        for (int i = 0; i < s.size(); ++i) {
            int curr = s[i] - 'a';
            if (index[curr].first == -1)
                index[curr].first = i;
            else
                index[curr].second = i;
        }
        int ans = 0;
        for (int i = 0; i < 26; ++i) {
            unordered_set<char> curr;
            for (int j = index[i].first + 1; j < index[i].second; ++j) {
                curr.insert(s[j]);
            }
            ans += curr.size();
        }
        return ans;
    }
};